woocommerce/woocommerce
# Add Short Description "Write With AI" Button + Inform about automatic generation

### Changes proposed in this Pull Request:

- Add a dedicated "Write With AI" button for the Short Product Description (excerpt) text field.
- When the Short Product Description is _automatically_ generated after generating the long description, show a small floating popup (using the Tour component) to inform the user that the short description was automatically generated. 
- Add a Woo AI plugin settings persistence layer for notices, tours, and more. This allows for dismissing the short description notice and to only show the notice if the specific notice has never been dismissed previously. 

Closes #39580 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to WooCommerce->Products->Add New, write a title, and generate a product description with Write With AI.
2. Note that the short description gets generated automatically and afterward, you see the popup notice.
3. Note that it closes when clicking "Got it".
4. Clear out the short description field and long description (or just create a new product post) and regenerate the long description.
5. Note that the short description gets automatically generated, but that you _do not_ see the popup notice.
6. With the short description still entered, use the Write With AI button to generate a short description. It should overwrite the existing.
7. Make sure that the short description Write With AI button cannot be clicked if the normal description is empty or less than 100 characters.



### Changelog entry




<details>

#### Significance




#### Type




#### Message [Woo AI] - Add a "Write with AI" button to the short description(excerpt) product editor text input.

#### Comment 

</details>
